test/functional/FunctionalBadProject.hs

```diff
@@ -2,37 +2,28 @@
 
 module FunctionalBadProject (tests) where
 
--- import           Control.Lens hiding (List)
--- import           Control.Monad.IO.Class
--- import qualified Data.Text as T
--- import           Language.LSP.Test hiding (message)
--- import           Language.LSP.Types as LSP
--- import           Language.LSP.Types.Lens as LSP hiding (contents, error )
+import           Control.Lens
+import qualified Data.Text                  as T
+import qualified Language.LSP.Protocol.Lens as L
 import           Test.Hls
+import           Test.Hls.Command
+
 
--- ---------------------------------------------------------------------
--- TODO: Currently this can not succeed, since such an error is thrown in "runActionWithContext" which
--- can produce diagnostics at the moment. Needs more investigation
--- TODO: @fendor: Add issue link here
---
 tests :: TestTree
-tests = testGroup "behaviour on malformed projects" [
-    testCase "no test executed" $ True @?= True
+tests = testGroup "behaviour on malformed projects"
```

Comment:
This is much more sophisticated than what I was thinking (although I imagine we'll want it anyway). I was just thinking of a unit test for the string matching function!

Reply:
Does that mean we don't want it?

Reply:
IDK, potentially good to have both?

Reply:
I think the tests are there and we shouldn't have them forever, any way. Ideally cabal gives us better error messages and hie-bios doesn't demand the parsing. Should be good enough, imo.